### PR TITLE
Feature 2 andriy melnyk

### PR DESCRIPTION
**Pull request by Melnyk9506**
Guys oleace check

### CODE REVIEW

File: server/src/main/java/com/lunchtime/models/Users.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import java.util.List;
+
+
+@Entity
+@Table(name = "users")
+@Data
+public class Users extends BaseEntity {
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/models/Users.java:17]: You should use singular noun in class name
https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names

File: server/src/main/java/com/lunchtime/models/Users.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import java.util.List;
+
+
+@Entity
+@Table(name = "users")
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/models/Users.java:15]: You should use singular noun in table name
https://rv-054java.slack.com/archives/CUKLK9279/p1583759766037900

File: server/src/main/java/com/lunchtime/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                          BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.userRepository = userRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+
+    @GetMapping("/all")
+    public ResponseEntity all() {
+        return ok(this.userRepository.findAll());
+    }
+//    @PostMapping("/sign-up")
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/UserController.java:28]: You should avoid useless comments in dev branch

File: server/src/main/java/com/lunchtime/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    private UserRepository userRepository;
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    public UserController(UserRepository userRepository,
+                          BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.userRepository = userRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+
+    @GetMapping("/all")
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/UserController.java:24]: You don't need to use ending 'all/ in get method. You can use standard REST approach

**Review by vitaliimak — CHANGES_REQUESTED** [review]
It isn't possible to create new User entity in system without POST method

**Comment by vitaliimak** [comment]
I can't run this code
```
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2020-03-16 20:32:58.320 ERROR 8803 --- [  restartedMain] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in com.lunchtime.controllers.UserController required a bean of type 'org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder' that could not be found.


Action:

Consider defining a bean of type 'org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder' in your configuration.
```

File: server/src/main/java/com/lunchtime/models/BaseEntity.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.Date;
+
+
+
+@MappedSuperclass
+@Data
+public class BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (Andrii35) [inline server/src/main/java/com/lunchtime/models/BaseEntity.java:24]: You should change IDENTITY to AUTO, becouse IDENTITY will be used with MySQL

File: server/src/main/java/com/lunchtime/models/Role.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import java.util.List;
+
+
+@Entity
+@Table(name = "roles")
+@Data
+public class Role extends BaseEntity {
+
```
Comment (Andrii35) [inline server/src/main/java/com/lunchtime/models/Role.java:18]: Should it not be here a primery key?

**Review by Andrii35 — COMMENTED** [review]
I run the project with docker. But by testing with Postman i became:
Could not get any response
There was an error connecting to http://localhost:8080/api/users/all.
Why this might have happened:
The server couldn't send a response:


Ensure that the backend is working properly

**Pull request closed without merging**